tests/models/llama/test_tokenization_llama.py

```diff
@@ -330,6 +331,15 @@ def test_add_prefix_space(self):
             fast_.decode(EXPECTED_WITH_SPACE, skip_special_tokens=True),
         )
 
+    def test_load_tokenizer_with_model_file_only(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            hf_hub_download(repo_id="huggyllama/llama-7b", filename="tokenizer.model", local_dir=tmp_dir)
+            tokenizer_fast = self.rust_tokenizer_class.from_pretrained(tmp_dir)
+            self.assertEqual(tokenizer_fast.encode("This is a test"), [1, 910, 338, 263, 1243])
+
+            tokenizer_slow = self.tokenizer_class.from_pretrained(tmp_dir)
+            self.assertEqual(tokenizer_slow.encode("This is a test"), [1, 910, 338, 263, 1243])
```

Comment:
Nice! Not sure I know how we have a slow tokenizer here that works 😄 as we don't have a general sentencepiece class

Reply:
this would be the LlamaTokenizer converter!

Reply:
but it is not supposed to work if you only downloaded a tokenizer.model file that is tiktoken b ased

Reply:
this `tokenizer.model` file would be sp model based

Reply:
although this PR enables the tiktoken PR https://github.com/huggingface/transformers/pull/31656,  it doesn't test / relate to tiktoken on its own. The tests here test that sp and fast models can be loaded without `config.json` which wasn't tested before, even though the code already supported it! 😄